js/packages/teams-ai/src/Application.ts

```diff
@@ -938,6 +962,59 @@ export class Application<TState extends TurnState = TurnState> {
             return handler(context);
         }
     }
+
+    /**
+     * If the user is signed in, get the access token. If not, triggers the sign in flow for the provided authentication setting name
+     * and returns. In this case, the bot should end the turn until the sign in flow is completed.
+     * @summary
+     * Use this method to get the access token for a user that is signed in to the bot.
+     * If the user isn't signed in, this method starts the sign-in flow.
+     * The bot should end the turn in this case until the sign-in flow completes and the user is signed in.
+     * @param {TurnContext} context - The context for the current turn with the user.
+     * @param {TState} state - The current state of the conversation.
+     * @param {string} settingName The name of the authentication setting.
+     * @returns {string | undefined} The token for the user if they are signed in, otherwise undefined.
+     */
+    public async getTokenOrStartSignIn(
```

Comment:
@blackchoey this is meant to be the method the end-user calls if they want to trigger sign in. It will put the user in "sign in" state and the subsequent incomming activities will be routed to the authentication logic in the `Application.run()` until sign in flow completes - either with a success or failure.